src/core/exprfilter.cpp

```diff
@@ -979,6 +1013,584 @@ do { \
+        vmaxps(x, x, ymmword_ptr[constants + ConstantIndex::exp_lo * 32]);
+        vmulps(fx, x, ymmword_ptr[constants + ConstantIndex::log2e * 32]);
+        vaddps(fx, fx, ymmword_ptr[constants + ConstantIndex::float_half * 32]);
+        vcvttps2dq(emm0, fx);
+        vcvtdq2ps(etmp, emm0);
+        vcmpps(mask, etmp, fx, _CMP_NLE_US);
+        vandps(mask, mask, one);
+        vsubps(fx, etmp, mask);
+        vfnmadd231ps(x, fx, ymmword_ptr[constants + ConstantIndex::exp_c1 * 32]);
+        vfnmadd231ps(x, fx, ymmword_ptr[constants + ConstantIndex::exp_c2 * 32]);
+        vmulps(z, x, x);
+        vmovaps(y, ymmword_ptr[constants + ConstantIndex::exp_p0 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::exp_p1 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::exp_p2 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::exp_p3 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::exp_p4 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::exp_p5 * 32]);
+        vfmadd213ps(y, z, x);
+        vaddps(y, y, one);
+        vcvttps2dq(emm0, fx);
+        vpaddd(emm0, emm0, ymmword_ptr[constants + ConstantIndex::x7F * 32]);
+        vpslld(emm0, emm0, 23);
+        vmulps(x, y, emm0);
+    }
+
+    void log_(YmmReg x, YmmReg zero, YmmReg one, Reg constants)
+    {
+        YmmReg emm0, invalid_mask, mask, y, etmp, z;
+        vcmpps(invalid_mask, zero, x, _CMP_NLE_US);
+        vmaxps(x, x, ymmword_ptr[constants + ConstantIndex::min_norm_pos * 32]);
+        vpsrld(emm0, x, 23);
+        vandps(x, x, ymmword_ptr[constants + ConstantIndex::inv_mant_mask * 32]);
+        vorps(x, x, ymmword_ptr[constants + ConstantIndex::float_half * 32]);
+        vpsubd(emm0, emm0, ymmword_ptr[constants + ConstantIndex::x7F * 32]);
+        vcvtdq2ps(emm0, emm0);
+        vaddps(emm0, emm0, one);
+        vcmpps(mask, x, ymmword_ptr[constants + ConstantIndex::sqrt_1_2 * 32], _CMP_LT_OS);
+        vandps(etmp, x, mask);
+        vsubps(x, x, one);
+        vandps(mask, mask, one);
+        vsubps(emm0, emm0, mask);
+        vaddps(x, x, etmp);
+        vmulps(z, x, x);
+        vmovaps(y, ymmword_ptr[constants + ConstantIndex::log_p0 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::log_p1 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::log_p2 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::log_p3 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::log_p4 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::log_p5 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::log_p6 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::log_p7 * 32]);
+        vfmadd213ps(y, x, ymmword_ptr[constants + ConstantIndex::log_p8 * 32]);
+        vmulps(y, y, z);
```

Comment:
I think after fmadding p8 there is a missing vmulps(y, y, x);
Plus there is a typo some lines below, float_half * 21 should be *32